docs/src/vector_transports.md

```diff
@@ -0,0 +1,48 @@
+## Vector transport
+
+Similar to the [exponential and logarithmic map](@ref exp-and-log) also the [parallel transport](@ref subsec-parallel-transport) might be costly to compute, especially when there is no closed form solution known and it has to be approximated with numerical methods.
+
+Similar to the [retraction and its inverse](@ref sec-retractions) the generalisation of the parallel transport can be phrased as follows
+
+A _vector transport_ is a way to transport a vector between two tangent spaces.
+Let ``p,q ∈ \mathcal M`` be given, ``c`` the curve along which we want to transport (cf. [parallel transport](@ref subsec-parallel-transport), for example a geodesic or a geodesic or curve given by a retraction, and ``X ∈ T_p\mathcal M`` be a tangent vector.
+Then ``T_{p→q}X`` is a vector transport if
+
+1. Consistency. ``T_{p→p}`` is the identiy on ``T_p\mathcal M``.
+2. Underlying curve ``T_{p→c(t)}X ∈ T_{c(t)}\mathcal M`` for ``t∈[0,1]``
+3. Linearity. ``T_{p→q}(X+Y)=T_{p→q}X + T_{p→q}Y``
```

Comment:
Where did you get these conditions from? Also, our scaled vector transport isn't linear as far as I'm aware.

Reply:
Its adapted from Absil Mahony Sepulchre, 2008, p. 170.
They introduce VT related to a retraction and basically define our `vector_transport_direction`, but where the realisation of the direction is done with a retraction (not necessarily with exp like for us).

So their retraction property I adapted to 2. consistency, but the other two are necessary. I have not checked whether scaled VT is a VT, but then maybe it is not ;)

Reply:
Hm, I'm not actually sure if you adapted the "associated retraction" condition properly. Why didn't you define `vector_transport_direction` as the fundamental one and `vector_transport_to` as the derived one? I know we currently have it in the code in the way it's in your docs but maybe it's the wrong way?

Reply:
You might be right – how would you adopt the to one? Or: Is there a proper way? Especially since we do it with exp (and not with any retract - which would be doable I think)?
I can check whether I see a VT definition for the to case as well (its just that AMS08 is my default reference for that).

Also - still - ScaledVT is not a VT according to that definition.

Reply:
This is quite problematic I think. Our current code is very much disconnected from the definition in AMS 2008 but that definition makes a lot of sense to me -- actually more then the design of our code. Having `vector_transport_direction` as the basic case makes sense, and we could have something like `associated_inverse_retraction(::AbstractManifold, ::AbstractVectorTransport)` that tells us how to get the `vector_transport_to` from `vector_transport_direction`. Also, we could have `AbstractVectorTransport` and `AbstractLinearVectorTransport` to accommodate both the AMS2008 definition and ScaledVT.

Reply:
Sure, sounds reasonable. We used `to` as a default since from parallel transport this is a default to use. But your redesign seems reasonable. Can you make a first proposal and I check?